spec/complex_number_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'complex_number'
+
+describe ComplexNumber do
+  
+  describe '#initialize' do
+    it 'will assign real and imaginary numbers' do
+    	c = ComplexNumber.new(2, 3)
+      expect(c.re).to eq 2
+      expect(c.im).to eq 3
+    end
+  end
+
+  # z -> z^2 + c
+
+  describe '#add' do
+  	it 'will add one number to another' do
+	  	# input
+	  	first = ComplexNumber.new(2, -3)
+	  	second = ComplexNumber.new(-3, 4)
+	  	
+	  	#expected result
+	  	expected = ComplexNumber.new(-1, 1)
+	  	result = first.add(second)
+	  	# (a + ib) + (x + iy) = (a + x) + i(b +y)
+	  	expect(result).to eq expected
+  	end
+  end
+
+  describe '#multiply' do
+  	it 'will multiply two numbers' do
+	  	# input
+	  	first = ComplexNumber.new(2, -3) # a, b
+	  	second = ComplexNumber.new(-3, 4) # x, y
+	  	
+	  	#expected result
+	  	# i * i = -1
+	  	#(a + ib)*(x + iy) = ax + iay + ibx - by
+	  	# re = ax - by = 2 * (-3) - (-3)*4 = -6 + 12 = 6
+	  	# i * (ay + bx) = 2*4 + (-3)*(-3) = 8 + 9 = 17  
+	  	mul = ComplexNumber.new(6, 17)
+	  	result = first.multiply(second)
+
+	  	expect(result).to eq mul 
+	  end 
+
+	  describe '#absolute_value_squared' do
+  	it 'will calculate the absolute value squared' do
+	  	# input
+	  	num = ComplexNumber.new(2, -3) # a, b
+	  	
+	  	#expected result
+	  	#mod**2 = a**2 + b**2
+
+	  	module_squared = 13
+	  	result = num.absolute_value_squared
+
+	  	expect(result).to eq module_squared 
+	  end
+	end 
+
+=begin --> Another way to do the module <--
+	describe '#absolute_value' do
+  	it 'will calculate the absolute value of a number' do
+	  	# input
+	  	num = ComplexNumber.new(2, -3) # a, b
+	  	
+	  	#expected result
+	  	#mod = sqrt(a**2 + b**2)
+
+	  	absolute_value = 3.605551275463989
+	  	result = num.absolute_value
+
+	  	expect(result).to be_within(0.00000001).of(absolute_value) 
+	  end
+	end 
+=end
+	
+	
+
+  end
+end
```

Comment:
This sign shows that you dont have a newline character at the end of the file. It is a good practice to have it, open the file, press Enter at the end, and save it.